lib/handleRequest.js

```diff
@@ -85,16 +85,23 @@ function preHandlerCallback (err, reply) {
   var result = reply.context.handler(reply.request, reply)
   if (result && typeof result.then === 'function') {
     result
-      .then((payload) => {
+      .then(function (payload) {
         // this is for async functions that
         // are using reply.send directly
         if (payload !== undefined || (reply.res.statusCode === 204 && reply.sent === false)) {
-          reply.send(payload)
+          // we use a try-catch internally to avoid adding a catch to another
+          // promise, increase promise perf by 10%
+          try {
+            reply.send(payload)
+          } catch (err) {
+            reply.sent = false
+            reply._isError = true
+            reply.send(err)
+          }
         } else if (reply.sent === false) {
           reply.res.log.error(new Error(`Promise may not be fulfilled with 'undefined' when statusCode is not 204`))
         }
-      })
-      .catch((err) => {
+      }, function (err) {
```

Comment:
Only a nit, could you use `catch` it's more verbose.

Reply:
The whole point of this PR is to remove `catch` to avoid creating another promise object.

Reply:
Oh 😸 , you're right I haven't seen it.